packages/web-components/fast-foundation/src/anchored-region/intersection-service.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ *
+ *
+ * @public
+ */
+export class IntersectionService {
+    private intersectionDetector: IntersectionObserver;
+
+    private observedElements: Element[] = [];
```

Comment:
You might consider using a [`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) instead of two arrays to manage element and observer callbacks. You can use the Element as a map key and the observer callback array as the value

Reply:
Yes, that is better, thx.